base/path.jl

```diff
@@ -163,17 +163,39 @@ See also: [`basename`](@ref)
 """
     basename(path::AbstractString) -> AbstractString
 
-Get the file name part of a path.
+Get the file/dir name part of a path.
+
+Generally, trailing '/' will be ignored. There're some other exceptions:
+
+* ""  ===> "."
```

Comment:
This is not what `basename` gives me on my system for empty input; I get empty output.

Reply:
Ah I see, it's documented that the command line utility can return either.